concrete/src/Entity/File/Version.php

```diff
@@ -893,8 +893,17 @@ public function getRelativePath()
         if (is_object($fsl)) {
             $configuration = $fsl->getConfigurationObject();
             if ($configuration->hasRelativePath()) {
-                return $configuration->getRelativePathToFile($cf->prefix($this->fvPrefix, $this->fvFilename));
+                $url = $configuration->getRelativePathToFile($cf->prefix($this->fvPrefix, $this->fvFilename));
             }
+            if (($configuration->hasPublicURL()) && (!$url)) {
```

Comment:
Here `$url` may not be set, causing a [PHP warning](https://3v4l.org/UYkEE).
Furthermore, there are a lot of extra parenthesis...